src/wp-includes/media.php

```diff
@@ -5564,6 +5680,110 @@ function wp_get_loading_attr_default( $context ) {
 	return 'lazy';
 }
 
+/**
+ * Get loading optimization attributes.
+ *
+ * @since 6.3.0
+ *
+ * @param string $tag_name The tag name.
+ * @param array[] $attr Array of the attributes for the tag.
+ * @param string $context Context for the element for which the loading optimization attribute is requested.
+ *
+ * @return array[] loading optimization attributes.
+ */
+function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
+	global $wp_query;
+
+	$loading_attrs = array();
+
+	// For now this function only supports images.
+	if ( 'img' !== $tag_name ) {
+		return $loading_attrs;
+	}
+
+	// For any resources, width and height must be provided, to avoid layout shifts.
+	if ( ! isset( $attr['width'] ) || ! isset( $attr['height'] ) ) {
+		return $loading_attrs;
+	}
+
+	// If the lazy-loading attribute already present then don't add `fetchpriority="high"`.
+	if ( isset( $attr['loading'] ) && 'lazy' === $attr['loading'] ) {
+		$loading_attrs['loading'] = 'lazy';
+		return $loading_attrs;
+	}
+
+	// If a 'fetchpriority' attribute with value 'high' is already provided, do not add `loading="lazy"`.
```

Comment:
```suggestion
	// An image with `fetchpriority="high"` cannot be assigned `loading="lazy"` at the same time.
```
Nit: "Why" rather than "What".